leakcanary-android/src/main/java/com/squareup/leakcanary/internal/DisplayLeakActivity.java

```diff
@@ -364,9 +376,29 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
         });
       }
       actionButton.setVisibility(leaks.size() == 0 ? GONE : VISIBLE);
+      shareButton.setVisibility(GONE);
     }
   }
 
+  private void shareLeakToStackOverflow() {
+    AnalyzedHeap visibleLeak = getVisibleLeak();
+    final String leakInfo = LeakCanary.INSTANCE.leakInfo(this, visibleLeak.heapDump, visibleLeak.result, false);
+    final Toast toast = Toast.makeText(this, R.string.leak_canary_leak_copied, Toast.LENGTH_SHORT);
+    final ClipboardManager clipboard = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
+    AsyncTask.execute(new Runnable() {
```

Comment:
Why is this an async task? Feels like we should ensure it's in the clip board before sending the intent.

I'm also confused by toast.show() being called from a background thread.

Reply:
I was running into a `StrictModeDiskWriteViolation` when the clipboard was set from the main thread. For the toast.show() part, I was also a little unsure of this as I wanted to ensure the toast was shown after the clipboard had been properly set, but then had to deal with this `AsyncTask`.